relay-server/src/services/processor.rs

```diff
@@ -889,6 +878,23 @@ pub struct ProcessEnvelope {
     pub reservoir_counters: ReservoirCounters,
 }
 
+/// Like a [`ProcessEnvelope`], but with an envelope which has been grouped.
+#[derive(Debug)]
+struct ProcessEnvelopeGroup {
```

Comment:
Nit: would suggest naming this `ProcessEnvelopeGrouped` or something like that. Currently it reads to me as "process this envelope group".